### PR TITLE
Audit

### PR DESCRIPTION
**Pull request by craig-iam-smith**
This is my result in the time constraints, focusing on security.  Additional tests need to be written to verify corner case functionality.

### CODE REVIEW

File: AuditGoalz-11Setp2024.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+**Auditor:** Craig Smith  
+**Date:** 11th September 2024  
+**Time constraint:** 2 hours exploratory audit (not a full audit)
+
+## Scope
+
+1) address Goalz.sol and GoalzToken.sol within this audit. 
+2) The mocks are out of scope for this audit.
+3) Gelato is out of scope for this audit.
+4) YieldGoalzUSD.sol is out of scope for this audit. (no tests exist)
```
Comment (mikeghen) [inline AuditGoalz-11Setp2024.md:12]: Yes, this can be removed afaik.

**Review by mikeghen — APPROVED** [review]
Thanks @craig-iam-smith, I will perform the remediations based on this.
